lib/core/proxy.c

```diff
@@ -456,8 +458,13 @@ static h2o_httpclient_body_cb on_head(h2o_httpclient_t *client, const char *errs
         h2o_iovec_t value = headers[i].value;
         if (h2o_iovec_is_token(headers[i].name)) {
             const h2o_token_t *token = H2O_STRUCT_FROM_MEMBER(h2o_token_t, buf, headers[i].name);
-            if (token->flags.proxy_should_drop_for_res)
+            if (token->flags.proxy_should_drop_for_res) {
+                if (token->flags.proxy_should_drop_for_res) {
```

Comment:
Isn't the check done in the surrounding block?

Reply:
Woah, thanks for noticing. Puzzled as to how this happened, but I will make no excuses, fixing!